python/sparkdl/graph/builder.py

```diff
@@ -67,17 +69,101 @@ def __exit__(self, *args):
     def run(self, *args, **kargs):
         """
         This method delegate the TensorFlow graph execution
-        to the underlying tf.Session object to perform 
+        to the underlying tf.Session object to perform
         one step of graph computation.
 
         All the parameters are defined according to `tf.Session.run`
-        Reference: https://www.tensorflow.org/api_docs/python/tf/Session#run        
+        Reference: https://www.tensorflow.org/api_docs/python/tf/Session#run
         """
         return self.sess.run(*args, **kargs)
 
+    def asUDF(self, udf_name, fetches, feeds_to_fields_map=None, blocked=False, register=True):
+        """
+        Register the graph in this session as a Spark SQL UserDefinedFunction.
+
+        The following example creates a UDF that takes the input
+        from a DataFrame column named 'image_col' and produce some random prediction.
+
+        .. code-block:: python
+
+            import sparkdl.graph.utils as tfx
+
+            with IsolatedSession(using_keras=True) as issn:
+                model = Sequential()
+                model.add(Flatten(input_shape=(640,480,3)))
+                model.add(Dense(units=64))
+                model.add(Activation('relu'))
+                model.add(Dense(units=10))
+                model.add(Activation('softmax'))
+                # Initialize the variables
+                init_op = tf.global_variables_initializer()
+                issn.run(init_op)
+                # Export the model as UDF
+                issn.asUDF('my_keras_model_udf',
+                           model.outputs,
+                           {tfx.op_name(issn.graph, model.inputs[0]): 'image_col'})
```

Comment:
Looks like we have to explicitly call `model.build()` to have the `model.input_names` available. https://github.com/fchollet/keras/blob/master/keras/models.py#L535